src/dict.h

```diff
@@ -68,14 +68,6 @@ typedef struct dictType {
     unsigned int keys_are_odd:1;
     /* TODO: Add a 'keys_are_even' flag and use a similar optimization if that
      * flag is set. */
-
-    /* Allow each dict and dictEntry to carry extra caller-defined metadata. The
-     * extra memory is initialized to 0 when allocated. */
-    size_t (*dictEntryMetadataBytes)(dict *d);
-    size_t (*dictMetadataBytes)(void);
-    /* Optional callback called after an entry has been reallocated (due to
-     * active defrag). Only called if the entry has metadata. */
-    void (*afterReplaceEntry)(dict *d, dictEntry *entry);
```

Comment:
One of my goals with this change is to unblock https://github.com/redis/redis/issues/10802 which would likely require different metadata format. I was thinking about removing existing metadata and then adding new appropriate format that will allow other memory efficiency improvements. I'm open for a discussion here though.

Reply:
I agree with @vitarb here. Let's get rid of that metadata feature.

Reply:
+1, unless we explicitly need it. Maybe we can do this revert it a second commit though so it's easy to revert if we need it?

Reply:
ok. i was thinking that this mechanism can be useful for something else in the future, but if it stands in the way of something else, we can drop it.
however, since these two mechanisms don't touch the same areas in the code, it would be easier to review the changes in this one, if we leave this revert for later.